core/domain/user_query_services.py

```diff
@@ -38,3 +40,30 @@ def save_new_query_model(
         query_status=feconf.USER_QUERY_STATUS_PROCESSING,
         user_ids=[]).put()
     return query_id
+
+
+def send_email_to_qualified_users(
+        query_id, email_subject, email_body, email_intent, email_option,
```

Comment:
This needs a docstring explaining what the args are. Please follow the style in e.g. core/domain/feedback_services.py.

Also, please use separate functions for the different email_option cases. Using the same function to handle multiple completely different modes is an antipattern.